gpioioctl/gpio.go

```diff
@@ -591,32 +595,85 @@ func (d *driverGPIO) After() []string {
 //
 // https://docs.kernel.org/userspace-api/gpio/chardev.html
 func (d *driverGPIO) Init() (bool, error) {
-	if runtime.GOOS == "linux" {
-		items, err := filepath.Glob("/dev/gpiochip*")
-		if err != nil {
-			return true, err
-		}
-		if len(items) == 0 {
-			return false, errors.New("no GPIO chips found")
+	Chips = make([]*GPIOChip, 0)
+	if runtime.GOOS != "linux" {
+		return true, nil
+	}
+	items, err := filepath.Glob("/dev/gpiochip*")
+	if err != nil {
+		return true, fmt.Errorf("gpioioctl: %w", err)
+	}
+	if len(items) == 0 {
+		return false, errors.New("no GPIO chips found")
+	}
+	// First, get all of the chips on the system.
+	chips := make([]*GPIOChip, 0)
+	var chip *GPIOChip
+	for _, item := range items {
+		chip, err = newGPIOChip(item)
+		if err == nil {
+			chips = append(chips, chip)
+		} else {
+			log.Println("gpioioctl.driverGPIO.Init() Error", err)
 		}
-		Chips = make([]*GPIOChip, 0)
-		for _, item := range items {
-			chip, err := newGPIOChip(item)
-			if err != nil {
-				log.Println("gpioioctl.driverGPIO.Init() Error", err)
-				return false, err
+	}
+	// Now, sort the chips so that those labeled with pinctrl- ( a Pi kernel standard)
+	// come first. Otherwise, sort them by label. This _should_ protect us from any
+	// random changes in chip naming/ordering.
+	sort.Slice(chips, func(i, j int) bool {
+		I := chips[i]
+		J := chips[j]
+		if I.Label()[:8] == "pinctrl-" {
```

Comment:
This will fail if a label is less than 8 characters. Use strings.HasPrefix() instead.

Reply:
Thanks.  I didn't see that.